### ISSUE TITLE
stream preparation - computing stream hydraulics

### ISSUE THREAD

**Issue by kavkapet** [storm-fsv-cvut/smoderp2d#55]
Podle mě si to nepřipojí data z DMT, vubec celej ten "stream" co si to udělá má nějak podezřele moc prázdných a diných polí. NODS na sebe nenavazují
[2Str_S_n100_t6_360mxx.zip](https://github.com/storm-fsv-cvut/smoem of input data. I suggest to include proper error message for user and leave it be...derp2d/files/3701277/2Str_S_n100_t6_360mxx.zip)
[5_hranice_toky_body.zip](https://github.com/storm-fsv-cvut/smoderp2d/files/3701284/5_hranice_toky_body.zip)

```
Computing stream hydraulics...
Failed script SMODERP2D2...

Traceback (most recent call last):
  File "D:\0_Smoderp\_2_D\0_skript\_git_CVUT\smoderp2d\bin\start-smoderp2d.py", line 13, in <module>
    sys.exit(runner.run())
  File "D:\0_Smoderp\_2_D\0_skript\_git_CVUT\smoderp2d\bin\..\smoderp2d\__init__.py", line 66, in run
    self._provider.load()
  File "D:\0_Smoderp\_2_D\0_skript\_git_CVUT\smoderp2d\bin\..\smoderp2d\providers\base\__init__.py", line 144, in load
    data = self._load_dpre()
  File "D:\0_Smoderp\_2_D\0_skript\_git_CVUT\smoderp2d\bin\..\smoderp2d\providers\arcgis\__init__.py", line 55, in _load_dpre
    return prep.run()
  File "D:\0_Smoderp\_2_D\0_skript\_git_CVUT\smoderp2d\bin\..\smoderp2d\providers\base\data_preparation.py", line 111, in run
    self._prepare_streams(mask_shp, dem_clip, intersect, flow_accumulation_clip)
  File "D:\0_Smoderp\_2_D\0_skript\_git_CVUT\smoderp2d\bin\..\smoderp2d\providers\base\data_preparation.py", line 495, in _prepare_streams
    self._streamPreparation(args)
  File "D:\0_Smoderp\_2_D\0_skript\_git_CVUT\smoderp2d\bin\..\smoderp2d\providers\arcgis\data_preparation.py", line 393, in _streamPreparation
    return StreamPreparation(args).prepare()
  File "D:\0_Smoderp\_2_D\0_skript\_git_CVUT\smoderp2d\bin\..\smoderp2d\providers\base\stream_preparation.py", line 62, in prepare
    self._stream_slope(stream)
  File "D:\0_Smoderp\_2_D\0_skript\_git_CVUT\smoderp2d\bin\..\smoderp2d\providers\arcgis\stream_preparation.py", line 253, in _stream_slope
    slope = (row[1] - row[2]) / row[4]
TypeError: unsupported operand type(s) for -: 'NoneType' and 'NoneType'
```

**Comment by landam** [comment]
@jerabekjak Any idea?

**Comment by jerabekjak** [comment]
This is problem of input data. I suggest to include proper error message for user and leave it be...

**Comment by kavkapet** [comment]
may be related to #66

**Comment by pesekon2** [comment]
I believe that this is solved since 15cd81be1e606b8147dd177c34359094f8c5318d. Closing the issue.
